### PR TITLE
C++: Make cpp/comparison-with-wider-type visible

### PR DESCRIPTION
**Pull request by jbj**
The results from this query look good on real-world projects, so let's make it visible by default.

Here is the list of all alerts from this query on lgtm.com: https://lgtm.com/rules/1505957036972/alerts/. The first alert (in spdk) is not great, but the other first few dozen alerts mostly LGTM.

### CODE REVIEW

**Comment by geoffw0** [comment]
Some of these look like they could be important results.

In many cases, something like `short` to `int` comparison is happening with values that are probably expected to always fit in the `short` - where one side has been 'optimized' to a smaller type whilst the other remains an `int`.  In some cases this may mean the code breaks under extreme conditions (e.g. when software rendering with a horizontal resolution of more than 32,767), in others it probably can't ever break (e.g. emulating an `enum`).  I see that the query already uses range analysis to reduce the rate of false positives, and in most cases if I were maintainer I would be happy to at least review each result we give and consider changing types.

It *is* a bit spammy on `scummvm/scummvm` though (210 results).

**Comment by jbj** [comment]
I agree it's spammy on ScummVM, but I'd also be surprised if none of those results are true positives. I'm sure it's fine to use 16-bit and 8-bit counters as they do as long as the input remains fixed -- historical data sets for LucasArts games -- but that doesn't mean we shouldn't give alerts for that sort of code.

File: change-notes/1.22/analysis-cpp.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 | No space for zero terminator (`cpp/no-space-for-terminator`) | Fewer false positive results | False positives involving strings that are not null-terminated have been excluded. |
 | Suspicious pointer scaling (`cpp/suspicious-pointer-scaling`) | Lower precision | The precision of this query has been reduced to "medium". This coding pattern is used intentionally and safely in a number of real-world projects. Results are no longer displayed on LGTM unless you choose to display them. |
 | Non-constant format string (`cpp/non-constant-format`) | Fewer false positive results | Rewritten using the taint-tracking library. |
+| Comparison of narrow type with wide type in loop condition (`cpp/comparison-with-wider-type`) | Higher precision | The precision of this query has been increased to "high" as the alerts from this query have proved to be valuable on real-world projects. With this precision, results are now displayed by default on LGTM. |
```
Comment (semmledocs-ac) [inline change-notes/1.22/analysis-cpp.md]:
```suggestion
| Comparison of narrow type with wide type in loop condition (`cpp/comparison-with-wider-type`) | Higher precision | The precision of this query has been increased to "high" as the alerts from this query have proved to be valuable on real-world projects. With this precision, results are now displayed by default in LGTM. |
```

**Comment by geoffw0** [comment]
Yes, and 210 results (which will be warnings) on scummvm isn't that bad in the context of 1.8k total alerts of which 755 are warnings (fairly high for a project this size).

Do the recent improvements to dataflow have any knock-on effects to range analysis and therefore this query?

I suspect we might be able to rule out some false positives from this query quite easily, e.g. by more tightly restricting to cases where the smaller-typed value appears to be updated inside the loop (in terms of the QL, make `loopVariant` narrower).

In any case I'm happy to merge this when the change note had been moved and updated.  We can always make improvements later on.

**Comment by jbj** [comment]
I'd rather let this PR wait until 1.23 has branched off.

**Comment by jbj** [comment]
Conflict resolved.

**Pull request merged**